exchanges/orderbook/tranches.go

```diff
@@ -78,27 +78,23 @@ updates:
 
 // deleteByID deletes reference by ID
 func (ts *Tranches) deleteByID(updts Tranches, bypassErr bool) error {
-updates:
 	for x := range updts {
-		for y := range *ts {
-			if updts[x].ID != (*ts)[y].ID {
-				continue
-			}
+		idx := slices.IndexFunc(*ts, func(t Tranche) bool {
```

Comment:
The changes here are a slight decrease in performance

```
master
 6487341	       197.3 ns/op	     480 B/op	       1 allocs/op

this branch:
 4987383	       236.9 ns/op	     480 B/op	       1 allocs/op
```

Reply:
I don't agree with the source of the delay being `IndexFunc`. I'm confident it's the Delete.

```
BenchmarkDeleteByID_Orig-8                       8801080               133.4 ns/op           480 B/op          1 allocs/op
BenchmarkDeleteByID_Delete-8                     7127386               167.2 ns/op           480 B/op          1 allocs/op
BenchmarkDeleteByID_IndexFunc-8                  8379580               143.3 ns/op           480 B/op          1 allocs/op
BenchmarkDeleteByID_IndexFuncDelete-8            6670990               179.2 ns/op           480 B/op          1 allocs/op
```

Reply:
Yeah in all my testing DeleteFunc was slower than an IndexFunc like I used in the other funcs which was quite hilarious. Whereas the origin wins in all circumstances (Ryan the 🏎️ driver)